bubblegum/program/README.md

```diff
@@ -1,4 +1,438 @@
+
+### Merkle proofs
+
+After an NFT is minted, for any operations that modify the NFT, Merkle proofs must be provided with the instruction to validate the Merkle tree changes.  Bubblegum is an Anchor program and makes use of the remaining accounts feature for this purpose.  Merkle proofs are added as remaining account `Pubkey`s that are 32-byte Keccak256 hash values that represent the nodes from the Merkle tree that are required to calculate a new Merkle root.
+
+### Creator verification
+
+Creators are specified in a creators array in the metadata passed into the `mint_v1` instruction.  All creators for which `verified` is set to true in the creators array at the time of mint must be a signer of the transaction or the mint will fail.  Beyond the signers specified in the `MintV1` Account validation structure, the `mint_v1` instruction uses remaining accounts to optionally look for additional signing creators.  This does not conflict with the Merkle proofs requirement because proofs are not required for minting.
+
+Beyond verifying creators at the time of mint, there are `verify_creator` and `unverify_creator` instructions that can be used on existing Compressed NFTs.
+
+### Collection verification
+
+Note that there is no such thing as compressed Verified Collections.  Collections are still NFTs created in the realm of token-metadata Metadata and Master Edition accounts.  Also note that a collection cannot be set to verified at the time of minting.  Instead, there are instructions to `verify_collection` and `unverify_collection`, as well as a `set_and_verify_collection` instruction for the case where the collection was set during the mint.  All of these require either the true Collection Authority to be a a signer, or a delegated Collection Authority to be a signer along with providing a Collection Authority Record PDA.  See the Metaplex documentation on [`Certified Collections`](https://docs.metaplex.com/programs/token-metadata/certified-collections) for more information on verifying collections.
+
+### Transfer ownership, delegate authority, and burn an NFT.
+
+Compressed NFTs support transferring ownership, delegating authority, and burning the NFT.  See the [Instructions](##Instructions) section below for details.
+
+### Redeem an NFT and decompress it into an uncompressed Metaplex NFT
+
+Redeeming an NFT removes the leaf from the Merkle tree and creates a voucher PDA account.  The voucher account can be sent to the `decompress_v1` instruction to decompress the NFT into an uncompressed Metaplex NFT.  As mentioned above this will cost rent for the token-metadata Metadata and Master Edition accounts that are created during the decompression process.  Note that after a compressed NFT is redeemed but before it is decompressed, the process can be reversed using `cancel_redeem`.  This puts the compressed NFT back into the Merkle tree.
```

Comment:
Same nit here with the double sequential "metadata" in the sentence.

Reply:
done